### PR TITLE
bump crypto to v0.22

### PR DESCRIPTION
**Pull request by semenov-vladyslav**
# Description of change

Bump iota-crypto to v0.22. Add Secp256k1EcdsaFlavor enum to primitives/procedures to distinguish between SHA256 and Keccak256 variants. Use BIP44 in cli example.

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .changes/secp256k1.md
Comment (thibault-martinez) [inline .changes/secp256k1.md:1]: Not fully sure you can reuse a .change file that has been consumed for a release already

File: client/src/tests/procedure_tests.rs [hunk 1]
```diff
@@ -398,24 +398,26 @@ async fn usecase_secp256k1() -> Result<(), Box<dyn std::error::Error>> {
...
         };
-        let mut sig_bytes: [u8; secp256k1_ecdsa::Signature::LENGTH] =
+        let mut sig_bytes: [u8; secp256k1_ecdsa::RecoverableSignature::LENGTH] =
             client.execute_procedure(secp256k1_ecdsa_sign).unwrap();
 
-        let sig = secp256k1_ecdsa::Signature::try_from_bytes(&sig_bytes).unwrap();
-        assert!(pk.verify(&sig, &msg));
-        assert_eq!(pk, sig.verify_recover(&msg).unwrap());
+        let sig = secp256k1_ecdsa::RecoverableSignature::try_from_bytes(&sig_bytes).unwrap();
+        // assert!(pk.verify(&sig, &msg));
```
Comment (thibault-martinez) [inline client/src/tests/procedure_tests.rs]: ?

File: .changes/secp256k1.md
Comment (semenov-vladyslav) [reply .changes/secp256k1.md:1]: If it was consumed why wasn't it removed?

Comment (thibault-martinez) [reply .changes/secp256k1.md:1]: Because it's a pre-release not a release. Honestly I don't really know covector that much, we can try like that...

Comment (Thoralf-M) [reply .changes/secp256k1.md:1]: It's added to the pre.json here https://github.com/iotaledger/stronghold.rs/blob/87d0ae07a26fc6711843174d1d706b066c85f350/.changes/pre.json#L4, so you need to remove it from there if you want covector to do something with it again

Comment (semenov-vladyslav) [reply .changes/secp256k1.md:1]: Lets hope it'll work

File: client/src/tests/procedure_tests.rs (hunk 1 above)
Comment (semenov-vladyslav) [reply client/src/tests/procedure_tests.rs]: Fixed. Forgot that we have `impl AsRef<Signature> for RecoverableSignature`.

File: .changes/secp256k1.md
Comment (thibault-martinez) [reply .changes/secp256k1.md:1]: Are you removing it from there then? Either that or create a new .change

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**